src/Parse/ParsePolygon.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Parse;
+
+use Parse\Internal\Encodable;
+use Parse\ParseGeoPoint;
+
+/**
+ * ParsePolygon - Representation of a Parse Polygon object.
+ *
+ * @author Diamond Lewis <findlewis@gmail.com>
+ */
+class ParsePolygon implements Encodable
+{
+    /**
+     * The coordinates.
+     *
+     * @var array
+     */
+    private $coordinates;
+
+    /**
+     * Create a Parse Polygon object.
+     *
+     * @param array $coords GeoPoints or Coordinates.
+     */
+    public function __construct($coords)
+    {
+        $this->setCoordinates($coords);
+    }
+
+    /**
+     * Set the Coordinates value for this Polygon.
+     *
+     * @param array $coords GeoPoints or Coordinates.
+     *
+     * @throws ParseException
+     */
+    public function setCoordinates($coords)
+    {
+        if (count($coords) < 3 || !is_array($coords)) {
+            throw new ParseException('Polygon must have at least 3 points');
+        }
+        $points = [];
+        foreach ($coords as $coord) {
+            $geoPoint = $coord;
+            if (is_array($coord)) {
```

Comment:
Since you're only adding this coordinate set if it's an array we should probably add another check here instead of silently skipping entries. Something that expects `$coord` to not only be an array, but to also contain a lat and long.